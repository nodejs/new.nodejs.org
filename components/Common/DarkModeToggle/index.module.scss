.darkModeToggle {
  background: none;
  border: none;
  color: var(--color-text-accent);
  cursor: pointer;
  line-height: 0;
  padding: 0;

  svg {
    font-size: 2rem;
  }
}
