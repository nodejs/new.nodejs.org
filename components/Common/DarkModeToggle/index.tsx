import { useTheme } from 'next-themes';
import { useIntl } from 'react-intl';
import { MdLightMode, MdNightlight } from 'react-icons/md';
import styles from './index.module.scss';

const DarkModeToggle = () => {
  const { theme, setTheme } = useTheme();

  const intl = useIntl();

  const isDark = theme === 'dark';

  const toggleTheme = (isKeyPress?: boolean) => {
    if (isKeyPress) {
      return;
    }

    setTheme(isDark ? 'light' : 'dark');
  };

  const ariaLabelText = intl.formatMessage({
    id: 'components.header.buttons.toggleDarkMode',
  });

  return (
    <button
      type="button"
      className={styles.darkModeToggle}
      onClick={() => toggleTheme()}
      onKeyDown={() => toggleTheme(true)}
      aria-pressed={isDark}
      aria-label={ariaLabelText}
    >
      <MdNightlight className="light-mode-only" />
      <MdLightMode className="dark-mode-only" />
    </button>
  );
};

export default DarkModeToggle;
