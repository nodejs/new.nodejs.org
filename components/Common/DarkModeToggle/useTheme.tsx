import { useContext } from 'react';
import { ThemeContext } from '../../../providers/themeprovider';

export function useTheme() {
  return useContext(ThemeContext) ?? [];
}
