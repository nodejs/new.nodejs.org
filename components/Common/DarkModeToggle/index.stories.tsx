import DarkModeToggle from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof DarkModeToggle>;
type Meta = MetaObj<typeof DarkModeToggle>;

export const Default: Story = {};

export default { component: DarkModeToggle } as Meta;
