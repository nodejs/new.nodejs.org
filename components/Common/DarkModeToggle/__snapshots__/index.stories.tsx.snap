// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common/DarkModeToggle Default smoke-test 1`] = `
<button type="button"
        class="DarkModeToggle_darkModeToggle__Mh_8x"
        aria-pressed="false"
        aria-label="Toggle dark/light mode"
>
  <svg stroke="currentColor"
       fill="currentColor"
       stroke-width="0"
       viewbox="0 0 24 24"
       class="light-mode-only"
       height="1em"
       width="1em"
       xmlns="http://www.w3.org/2000/svg"
  >
    <path fill="none"
          d="M0 0h24v24H0V0z"
    >
    </path>
    <path d="M14 2c1.82 0 3.53.5 5 1.35-2.99 1.73-5 4.95-5 8.65s2.01 6.92 5 8.65A9.973 9.973 0 0114 22C8.48 22 4 17.52 4 12S8.48 2 14 2z">
    </path>
  </svg>
  <svg stroke="currentColor"
       fill="currentColor"
       stroke-width="0"
       viewbox="0 0 24 24"
       class="dark-mode-only"
       height="1em"
       width="1em"
       xmlns="http://www.w3.org/2000/svg"
  >
    <path fill="none"
          d="M0 0h24v24H0z"
    >
    </path>
    <path d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0a.996.996 0 000-1.41l-1.06-1.06zm1.06-10.96a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z">
    </path>
  </svg>
</button>
`;
